<?xml version="1.0"?>
<launch>
    <arg name="sim" value="true"/>
    <arg name="gui" default="false"/>
    <arg name="reverse_omega" default="false"/>
    <arg name="use_fake_local_planner" default="false" />
    
    <include file="$(find docking_sim)/launch/link.launch">
        <arg name="use_fake_local_planner" value="$(arg use_fake_local_planner)" />
        <arg name="gui" value="$(arg gui)"/>
    </include>
    
    <node pkg="rap_controller" type="rap_controller.py" name="rap_controller_car1" output='screen'>
        <param name="robot_name" value="car1"/>
        <param name="role" value="leader"/>
        <param name="sim" value="$(arg sim)"/>
        <param name="ctl_frequency" value="30"/>
        <param name="reverse_omega" value="$(arg reverse_omega)"/>
        <param name="map_frame" value="map"/>
        <param name="base_link_frame" value="car1"/>
        <param name="big_car_frame" value="base_link"/>
        <param name="cmd_vel_topic" value="/cmd_vel_car1"/>
        <remap from="/car1/rap_cmd" to="/cmd_vel"/>
    </node>
    
    <node pkg="rap_controller" type="rap_controller.py" name="rap_controller_car2" output='screen'>
        <param name="robot_name" value="car2"/>
        <param name="role" value="follower"/>
        <param name="sim" value="$(arg sim)"/>
        <param name="reverse_omega" value="$(arg reverse_omega)"/>
        <param name="ctl_frequency" value="30"/>
        <param name="map_frame" value="map"/>
        <param name="base_link_frame" value="car2"/>
        <param name="big_car_frame" value="base_link"/>
        <param name="cmd_vel_topic" value="/cmd_vel_car2"/>
        <remap from="/car2/rap_cmd" to="/cmd_vel"/>
    </node>
</launch>
